### DISCUSSION TITLE
Variable to Blade

### DISCUSSION THREAD

**Discussion by MrBaker10** [ismaelw/laratex#9] [General]
Hey, 
sorry for the probably simple question. However, I am relatively new to php and laravel.

I have a query.

`$items = Benefit::searchBy($param)->orderBy('Buchung', 'desc')->get();`

Now I would like to get the variable "$items" into the document. How do I get the data into the blade?

In the blade I have a table with a foreach loop.
             
             @foreach($items as $key => $item)
                {{ $key }} & {{ $item }} \\ \hline
             @endforeach

**Comment by ismaelw** [comment]
Hi @MrBaker10 

No worries. I will try to help you. But right now I don't have enough information.
Is it possible to tell me what the `$items` contains?

Also, please show me the function that sends the data to LaraTeX to render it using your blade. 

Kind regards,
Ismael

**Comment by MrBaker10** [comment]
$items is a collection based on what the user is looking for.

`$param = $request->session()->get('param');`
`$items = Benefit::searchBy($param)->orderBy('Buchung', 'desc')->get();`

The data comes from the model Benefit.

The function is only slightly modified from your example function.

` return (new LaraTeX('latex.tex'))->with([
            'Name' => 'John Doe',
            'Dob' => '01/01/1990',
            'SpecialCharacters' => '$ (a < b) $',
            'items' => $items
        ])->download($filename);`

**Reply by ismaelw** [reply]
And what data is inside the `$items` variable? I don't know what you need to output in your blade after.

How I always do it is to use the PHP method `compact` to generate an array of all my variables that I want to pass to the view.
```
$param = $request->session()->get('param');
$name = 'John Doe';
$items = Benefit::searchBy($param)->orderBy('Buchung', 'desc')->get();
new LaraTeX('blade')->with(compact('items', 'name'))
```

Make sure that the variables `items` and `name` from my example here exist in your context. 
In that way I can use the variable `$name` for example in my blade file or do a simple foreach over the `$items` collection:

```
@foreach ($items as $item)
    <p>This is items ID: {{ $item->id }}</p>
@endforeach
```

Can you try if that works for your items collection?

**Reply by ismaelw** [reply]
Did you already found time to look into it? Did you already make some progress?

**Comment by MrBaker10** [comment]
Yeah thanks you it works.

Sorry for the late response!

**Reply by ismaelw** [reply]
No worries! I am glad that it works now!
